stl/inc/xstring

```diff
@@ -2468,6 +2468,69 @@ public:
         _Proxy._Release();
     }
 
+    basic_string(_String_constructor_concat_tag, basic_string& _Left, basic_string& _Right)
+        : _Mypair(_One_then_variadic_args_t(), _Left._Getal()) {
+        auto& _My_data    = _Mypair._Myval2;
+        auto& _Left_data  = _Left._Mypair._Myval2;
+        auto& _Right_data = _Right._Mypair._Myval2;
+        _Left_data._Orphan_all();
+        _Right_data._Orphan_all();
+        const auto _Left_size  = _Left_data._Mysize;
+        const auto _Right_size = _Right_data._Mysize;
+
+        const auto _Left_capacity  = _Left_data._Myres;
+        const auto _Right_capacity = _Right_data._Myres;
+        // overflow is OK due to max_size() checks:
+        const auto _New_size = static_cast<size_type>(_Left_size + _Right_size);
+        if (_Allocators_equal(_Getal(), _Right._Getal()) && _Right_capacity > _Left_capacity

```

Comment:
Am I confused, or should the `_Left` case be first since it avoids the extra cost of shifting to the end when both allocators are equal (the common case)? Sure, `_Right_capacity > _Left_capacity` suggests that we're preserving the larger allocation, which is nice, but it will also correlate strongly with `_Right_size > _Left_size`.

Reply:
The case order doesn't matter: the point is to keep the larger existing allocation. I guess I can put the left one first just for "move the hottest part of the function toward the beginning" reasons but it won't change which of these two branches is engaged.

Reply:
I don't see the potential future benefit of keeping the larger allocation as more valuable than performing fewer character copies _now_, but I suspect there's no way to resolve the difference if you disagree. We can't project which choice is the best without knowing what the program will do with these strings.

Reply:
In this case it's less "I disagree" and more "I'm doing what the old code was doing".

Reply:
(And as a result I guess I disagree in the sense that that's a performance tuning / variable I do not think we should tune/change right now without data showing what's better)

Reply:
You had me at "this is what the old code was doing", which hadn't occurred to me. (Yeah, I know this is obvious - but it's all green!) We shouldn't change this behavior without a much better reason than "I have a hunch this should be different."